src/bot.ts

```diff
@@ -448,3 +450,29 @@ you can circumvent this protection against memory leaks.`);
         this.errorHandler = errorHandler;
     }
 }
+
+async function withRetries(task: () => Promise<unknown>) {
+    let success = false;
+    while (!success) {
+        try {
+            await task();
+            success = true;
```

Comment:
i'd return the result of `task()` here, so the function is more useful for the future :P

Reply:
Good point

Reply:
Good point!

Reply:
This will be done in #182.